BASS Audio Test/ObjectiveBASS.m

```diff
@@ -983,39 +982,43 @@ - (NSError *)errorForErrorCode:(BassStreamError)erro {
 
 #pragma mark - Audio Session Routing/Interruption Handling
 
-- (void)setupAudioSession:(BOOL)addObservers {
+- (void)setupAudioSession:(BOOL)makeActive {
     AVAudioSession *session = AVAudioSession.sharedInstance;
     
     [session setCategory:AVAudioSessionCategoryPlayback
              withOptions:AVAudioSessionCategoryOptionAllowAirPlay | AVAudioSessionCategoryOptionAllowBluetooth
                    error:nil];
     
-    [session setActive:YES
-                 error:nil];
+    if (makeActive) {
+        [session setActive:YES
+                     error:nil];
+    }
+    
+    // Register for Route Change notifications
+    if (makeActive && !audioSessionObserversSetUp) {
```

Comment:
This is an unrelated speculative bug that I added a fix for. If a session gets set up while audio is playing then the observers will never get added if the other audio stops playing later (or if the app initiates playback of its own audio)